angular/src/providers/angular-delegate.ts

```diff
@@ -165,3 +182,7 @@ const getProviders = (params: { [key: string]: any }) => {
 const provideNavParamsInjectable = (params: { [key: string]: any }) => {
   return new NavParams(params);
 };
+
+function isComponentFactoryResolver(item: any): item is ComponentFactoryResolver {
+  return !!item.resolveComponentFactory;
+}
```

Comment:
This function is duplicated in `ion-router-outlet.ts`; would it be possible to make a shared util instead?